### PR TITLE
add jsk_footprint package

### PR DESCRIPTION
**Pull request by aginika**
I included one executable which will reconfigure the move_base local_costmap footprint in this package.

pr2_basefoot2

### CODE REVIEW

File: jsk_footprint/package.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0"?>
+<package>
+  <name>jsk_footprint</name>
+  <version>0.0.0</version>
```
Comment (garaemon) [inline jsk_footprint/package.xml]: Fix version number

File: jsk_footprint/package.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  <description>The jsk_footprint package</description>
+
+  <maintainer email="inagaki@todo.todo">inagaki</maintainer>
```
Comment (garaemon) [inline jsk_footprint/package.xml]: fix email address

File: jsk_footprint/package.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  <license>TODO</license>
```
Comment (garaemon) [inline jsk_footprint/package.xml]: fix license

File: jsk_footprint/src/costmap_footorint_reconfigure.cpp [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+        consider_robotarms_ = true;
+      }
+
+    if (!n_.getParam("consider_cloud", consider_cloud_))
+      {
+        ROS_WARN("~consider_cloud is not specified");
+        consider_cloud_ = false;
+      }
+
+    if (!n_.getParam("default_footprint_string", default_footprint_string_))
```
Comment (garaemon) [inline jsk_footprint/src/costmap_footorint_reconfigure.cpp]: Please use jsk_topic_tools/rosparam_utils.h

File: jsk_footprint/src/costmap_footorint_reconfigure.cpp [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+      if (!n_.getParam("tf_names", tmp_tf_names))
+        {
+          ROS_WARN("~tf_names is not specified");
+          tmp_tf_names = std::string("/larm,/rarm");
+        }
+      boost::algorithm::split( tf_names_, tmp_tf_names, boost::is_any_of(","));
+    }
+    
+    srv_ = n_.advertiseService("reconfigure_costmap_footprint", &CostmapFootPrintReconfigure::request_cloud, this);
+    sub_ = n_.subscribe("input", 1, &CostmapFootPrintReconfigure::cloud_cb,this);
```
Comment (garaemon) [inline jsk_footprint/src/costmap_footorint_reconfigure.cpp]: What is ~input pointcloud?
if the pointcloud is large like kinect/stereo pointcloud, please implement it as nodelet.

File: jsk_footprint/src/costmap_footorint_reconfigure.cpp [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+    if(enable_box_publish_){
+      pub_ = n_.advertise<jsk_pcl_ros::BoundingBoxArray>("boxes", 1);
+      pub2_ = n_.advertise<jsk_pcl_ros::BoundingBox>("box", 1);
+    }
+  }
+
+  void
+  cloud_cb (const sensor_msgs::PointCloud2 &pc){
```
Comment (garaemon) [inline jsk_footprint/src/costmap_footorint_reconfigure.cpp]: no space before paren according to your coding style

File: jsk_footprint/src/costmap_footorint_reconfigure.cpp [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+      double_param.value = result_string;
+    }
+    conf.strs.push_back(double_param);
+    if(enable_callrequest_){
+      srv_req.config = conf;
+      ros::service::call("/move_base_node/local_costmap/set_parameters", srv_req, srv_resp);
+    }
+  }
+
+  std::string calcConvexHull()
```
Comment (garaemon) [inline jsk_footprint/src/costmap_footorint_reconfigure.cpp]: camel case? snake case?

File: jsk_footprint/src/costmap_footorint_reconfigure.cpp [hunk 8]
```diff
@@ -0,0 +1,341 @@
...
+          tf::StampedTransform transform;
+          tf_listener.waitForTransform( basic_coordiname_tf_, tf_name, ros::Time(0), ros::Duration(0.5) );
+          tf_listener.lookupTransform( basic_coordiname_tf_, tf_name, ros::Time(0), transform);
+          pcl::PointXYZ tmp_point;
+          tmp_point.x = transform.getOrigin().x();
+          tmp_point.y = transform.getOrigin().y();
+          tmp_point.z = 0;//transform.getOrigin().z();
+          default_points->points.push_back(tmp_point);
+        } catch (tf::TransformException ex) {
+          ROS_INFO("%s",ex.what());
```
Comment (garaemon) [inline jsk_footprint/src/costmap_footorint_reconfigure.cpp]: ROS_ERROR or ROS_WARN is preferred in this case

File: jsk_footprint/src/costmap_footorint_reconfigure.cpp [hunk 9]
```diff
@@ -0,0 +1,341 @@
...
+        {
+          ROS_WARN("~tf_names is not specified");
+          tmp_tf_names = std::string("/larm,/rarm");
+        }
+      boost::algorithm::split( tf_names_, tmp_tf_names, boost::is_any_of(","));
+    }
+    
+    srv_ = n_.advertiseService("reconfigure_costmap_footprint", &CostmapFootPrintReconfigure::request_cloud, this);
+    sub_ = n_.subscribe("input", 1, &CostmapFootPrintReconfigure::cloud_cb,this);
+    if(enable_box_publish_){
```
Comment (garaemon) [inline jsk_footprint/src/costmap_footorint_reconfigure.cpp]: advretise before subscribing any topics

**Comment by garaemon** [comment]
I don't know this code should have new package or go to `jsk_pcl_ros`.
Please let me knows pros/cons about creating this new package.

**Comment by aginika** [comment]
I wondered that too.
I sent this PR to hear about where I should send PR.
This program is mainly to reconfigure the footprint with arms and base or pointcloud, so I felt this seems to be odd in jsk_pcl_ros?
But now this program also publish the BoundingBox, so this may be fine to be in jsk_pcl_ros.
I don't have the obvious pros, so should I send to jsk_pcl_ros?

**Comment by garaemon** [comment]
> This program is mainly to reconfigure the footprint with arms and base or pointcloud, so I felt this seems to be odd in jsk_pcl_ros?

pointcloud of footprint?

**Comment by aginika** [comment]
I mean this will consider some pointcloud into footprint if you want, such as grasp object pointcloud.

**Comment by garaemon** [comment]
> I mean this will consider some pointcloud into footprint if you want, such as grasp object pointcloud.

Hmm, I think `jsk_pcl_ros` would be better according to what you said.
Because this program should be implemented as nodelet, and I don't want to manage too many nodelet packages.

**Comment by aginika** [comment]
Ok, thank you for your advise. I will send this to jsk_pcl_ros

**Pull request closed without merging**